mmv1/third_party/terraform/website/docs/guides/using_write_only_attributes.html.markdown

```diff
@@ -0,0 +1,156 @@
+  secret_data_wo_version = 1
+  enabled = true
+}
+```
+
+During `terraform plan` you will see that the write-only attribute is marked appropriately:
+
+```
+  # google_secret_manager_secret_version.secret-version-basic will be created
+  + resource "google_secret_manager_secret_version" "secret-version-basic" {
+      + create_time            = (known after apply)
+      + deletion_policy        = "DELETE"
+      + destroy_time           = (known after apply)
+      + enabled                = true
+      + id                     = (known after apply)
+      + is_secret_data_base64  = false
+      + name                   = (known after apply)
+      + secret                 = (known after apply)
+      + secret_data_wo         = (write-only attribute)
+      + secret_data_wo_version = 1
+      + version                = (known after apply)
+    }
+```
+
+Upon `terrform apply` you will see in `terraform.tfstate` that the write-only attribute from the configuration is not reflected in the state:
+
+```hcl
+...
+    {
+      "mode": "managed",
+      "type": "google_secret_manager_secret_version",
+      "name": "secret-version-basic",
+      "provider": "provider[\"registry.terraform.io/hashicorp/google\"]",
+      "instances": [
+        {
+          "schema_version": 0,
+          "attributes": {
+            "create_time": "2025-02-19T05:56:20.942452Z",
+            "deletion_policy": "DELETE",
+            "destroy_time": "",
+            "enabled": true,
+            "id": "projects/871647908372/secrets/secret-version/versions/1",
+            "is_secret_data_base64": false,
+            "name": "projects/871647908372/secrets/secret-version/versions/1",
+            "secret": "projects/hc-terraform-testing/secrets/secret-version",
+            "secret_data": null,
+            "secret_data_wo": null,
+            "secret_data_wo_version": 1,
+            "timeouts": null,
+            "version": "1"
+          },
+          ...
+        }
+      ]
+    }
+```
+
+Any value that is set for a write-only attribute is nulled out before the RPC response is sent to Terraform.
+
+### Updating write-only attributes
+
+Since write-only attributes are not stored in the Terraform state, they cannot be updated by just changing the value in the configuration due to the attribute being nulled out.
+
+In order to update a write-only attribute we must change the write-only attribute's version.
+
+```hcl
+resource "google_secret_manager_secret" "secret-basic" {
+  secret_id = "secret-version"
+
+  labels = {
+    label = "my-label"
+  }
+
+  replication {
+    auto {}
+  }
+}
+
+
+resource "google_secret_manager_secret_version" "secret-version-basic" {
+  secret = google_secret_manager_secret.secret-basic.id
+  secret_data_wo = "new-secret-data"
+  secret_data_wo_version = 2
+  enabled = true
+}
+```
+
+A `terraform apply` of this configuration will allow you to update the write-only attribute as it will destroy and recreate the resource.
```

Comment:
I would recommend using a different resource as your example, one that can be updated. This teaches a first-time reader that updating a WO attribute destroys their resource when it only does if the corresponding sensitive value was immutable!

Reply:
used `password_wo` to showcase updating the write-only where the resource isn't immutable.